server/etc/rc.d/init.d/pulp_streamer

```diff
@@ -0,0 +1,208 @@
+#!/bin/sh
+# =============================================================
+#  pulp_streamer - Starts Pulp's Lazy Content Streamer service.
+# =============================================================
+#
+# :Usage: /etc/init.d/pulp_streamer {start|stop|force-reload|restart|try-restart|status}
+# :Configuration file: /etc/default/pulp_streamer
+
+### BEGIN INIT INFO
+# Provides:          pulp_streamer
+# Required-Start:    $network $local_fs $remote_fs
+# Required-Stop:     $network $local_fs $remote_fs
+# Should-Start:      mongod qpidd rabbitmq-server
+# Default-Start:     3 4 5
+# Default-Stop:      0 1 2 6
+# Short-Description: pulp's lazy content streamer service
+### END INIT INFO
+
+TWISTD="/usr/bin/twistd"
+TWISTD_USER="apache"
+TWISTD_GROUP="apache"
+TWISTD_PID_FILE="/var/run/pulp/pulp_streamer.pid"
+TWISTD_PID_DIR="/var/run/pulp"
+TWISTD_LOG_FILE="/var/log/pulp/pulp_streamer.log"
+TWISTD_LOG_DIR="/var/log/pulp"
+TWISTD_OPTS="--pidfile=$TWISTD_PID_FILE --syslog --prefix=pulp_streamer --python /srv/pulp/streamer.tac"
+
+export PATH="${PATH:+$PATH:}/usr/sbin:/sbin"
+
+if [ -L "$0" ]; then
+    SCRIPT_FILE=$(readlink "$0")
+else
+    SCRIPT_FILE="$0"
+fi
+SCRIPT_NAME="$(basename "$SCRIPT_FILE")"
+
+if [ "$EUID" != "0" ]; then
+    echo "Error: This program can only be used by the root user."
+    exit 1
+fi
+
+if test -f /etc/default/pulp_streamer; then
+    write_log_message "Loading config from /etc/default/pulp_streamer"
+    scripts="/etc/default/pulp_streamer"
+    . /etc/default/pulp_streamer
+fi
+
+start_streamer() {
+    echo "Starting ${SCRIPT_NAME}..."
+    su - "$TWISTD_USER" -s /bin/sh -c "$TWISTD $TWISTD_OPTS"
+    echo "OK"
+}
+
+stop_streamer () {
+    echo -n "Stopping ${SCRIPT_NAME}... "
+    if [ -f "$TWISTD_PID_FILE" ]; then
+        wait_pid $(cat "$TWISTD_PID_FILE")
+    else
+        echo "NOT RUNNING"
+    fi
+}
+
+check_dev_null() {
+    if [ ! -c /dev/null ]; then
+        echo "/dev/null is not a character device!"
+        exit 75  # EX_TEMPFAIL
+    fi
+}
+
+maybe_die() {
+    if [ $? -ne 0 ]; then
+        echo "Exiting: $*"
+        exit 77  # EX_NOPERM
+    fi
+}
+
+create_default_dir() {
```

Comment:
I agree, though I just took what the pulp_celerybeat upstart script was doing and modified it. I expect we can track down an Upstart master to help us write better scripts.